climada/util/coordinates.py

```diff
@@ -891,6 +892,70 @@ def mapping_grid2flattened(col, row, matrix_shape):
         raise ValueError
     return row * matrix_shape[1] + col
 
+def assign_coordinates(coords, coords_to_assign, method="NN", distance="haversine", threshold=100):
```

Comment:
It is starting to getting an old question, sorry for repeating it again: do we handle the antimeridian correctly here or is it up to the code/user to first transform the data?

Reply:
The exact matching will only work if all longitudinal values are in the same longitudinal range. However, the "haversine" distance is able to correctly find nearest neighbors independent of longitudinal range. I added a unit test for that and added a note in the docstring.